diff -Naur gnome-packagekit-2.32.0.pristine/data/gnome-packagekit.schemas.in gnome-packagekit-2.32.0.new/data/gnome-packagekit.schemas.in
--- gnome-packagekit-2.32.0.pristine/data/gnome-packagekit.schemas.in	2010-09-22 17:20:31.000000000 +0200
+++ gnome-packagekit-2.32.0.new/data/gnome-packagekit.schemas.in	2010-10-12 14:47:30.658605449 +0200
@@ -324,7 +336,7 @@
       <applyto>/apps/gnome-packagekit/update-icon/frequency_get_upgrades</applyto>
       <owner>gnome-packagekit</owner>
       <type>int</type>
-      <default>604800</default>
+      <default>0</default>
       <locale name="C">
         <short>How often to check for distribution upgrades</short>
         <long>How often to check for distribution upgrades. Value is in seconds.</long>
@@ -540,7 +552,7 @@
       <applyto>/apps/gnome_settings_daemon/gtk-modules/pk-gtk-module</applyto>
       <owner>gnome-packagekit</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
         <short>GTK+ module for font installation</short>
         <long>This key determines if applications should be able to prompt for fonts</long>
diff -Naur gnome-packagekit-2.32.0.pristine/data/gpk-install-file.desktop.in gnome-packagekit-2.32.0.new/data/gpk-install-file.desktop.in
--- gnome-packagekit-2.32.0.pristine/data/gpk-install-file.desktop.in	2010-09-22 17:20:47.000000000 +0200
+++ gnome-packagekit-2.32.0.new/data/gpk-install-file.desktop.in	2010-10-12 14:48:40.332791118 +0200
@@ -8,5 +8,5 @@
 Icon=system-software-install
 StartupNotify=true
 NoDisplay=true
-MimeType=application/x-rpm;application/x-redhat-package-manager;application/x-servicepack;application/x-deb;
+MimeType=application/x-compressed-tar;application/x-xz-compressed-tar;application/x-servicepack;
 
