diff -Naur mintdesktop.orig/usr/lib/linuxmint/mintDesktop/mintDesktop.py mintdesktop/usr/lib/linuxmint/mintDesktop/mintDesktop.py
--- mintdesktop.orig/usr/lib/linuxmint/mintDesktop/mintDesktop.py	2011-03-07 23:02:34.000000000 +0100
+++ mintdesktop/usr/lib/linuxmint/mintDesktop/mintDesktop.py	2011-03-07 23:02:49.255148919 +0100
@@ -175,7 +175,7 @@
 		dlg.set_program_name("mintDesktop")	
 		dlg.set_comments(_("Desktop Settings"))
 		try:
-			h = open('/usr/share/common-licenses/GPL','r')
+			h = open('/usr/share/licenses/common/GPL/license.txt','r')
 			s = h.readlines()
 			gpl = ""
 			for line in s:
@@ -186,7 +186,7 @@
 			print detail
 		try: 
 			version = commands.getoutput("/usr/lib/linuxmint/common/version.py mintdesktop")
-			dlg.set_version(version)
+			dlg.set_version("3.2.2")
 		except Exception, detail:
 			print detail
 
