diff -Naur ocrfeeder-0.7.6.orig/src/ocrfeeder/studio/studioBuilder.py ocrfeeder-0.7.6/src/ocrfeeder/studio/studioBuilder.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/studio/studioBuilder.py	2011-08-03 23:58:05.000000000 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/studio/studioBuilder.py	2011-10-06 06:37:39.540195649 +0200
@@ -422,8 +422,6 @@
         self.main_window.setHasImages(not is_empty)
 
     def about(self, widget = None):
-        from gnome import url_show
-        gtk.about_dialog_set_url_hook(lambda x,y,z: url_show(y), "")
         about_dialog = widgetPresenter.CustomAboutDialog()
         if about_dialog.run():
             about_dialog.destroy()
diff -Naur ocrfeeder-0.7.6.orig/src/ocrfeeder/util/lib.py ocrfeeder-0.7.6/src/ocrfeeder/util/lib.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/util/lib.py	2011-10-06 05:54:25.000000000 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/util/lib.py	2011-10-06 06:37:37.110345885 +0200
@@ -25,7 +25,6 @@
 import gtk
 import math
 from constants import *
-from gnome import url_show
 import sane
 import tempfile
 
@@ -92,9 +91,6 @@
         i += 1
     return name + str(i) + ext
 
-def openUrl(widget, url, data):
-    url_show(url)
-
 def getStandardDeviation(list_of_values):
     if not list_of_values:
         return 0
