=== modified file 'configure.ac'
--- configure.ac	2010-11-10 15:44:40 +0000
+++ configure.ac	2010-11-18 12:25:56 +0000
@@ -9,7 +9,7 @@
 AM_MAINTAINER_MODE
 
 IT_PROG_INTLTOOL([0.35.0])
-AM_PROG_VALAC([0.7.8])
+AM_PROG_VALAC([0.11.2])
 AS_IF([test -z "$VALAC"], [AC_MSG_ERROR(["No valac compiler found."])])
 
 AC_ISC_POSIX
=== modified file 'src/fetch-file.vala'
--- src/fetch-file.vala	2010-09-01 16:51:30 +0000
+++ src/fetch-file.vala	2010-11-18 11:30:03 +0000
@@ -64,7 +64,7 @@
     ssize_t bufsize = 1;
     do {
       try {
-        bufsize = yield this.stream.read_async (buffer, size, GLib.Priority.DEFAULT, null);
+        bufsize = yield this.stream.read_async (buffer, GLib.Priority.DEFAULT, null);
         if (bufsize < 1) { break;}
 
         if (bufsize != size)

=== modified file 'src/metadata-menu-item.vala'
--- src/metadata-menu-item.vala	2010-09-15 11:30:06 +0000
+++ src/metadata-menu-item.vala	2010-11-18 11:30:03 +0000
@@ -133,7 +133,7 @@
 	{
 		try{
 			PixbufLoader loader = new PixbufLoader ();
-			loader.write (update.data, update.len);
+			loader.write (update.data);
 			loader.close ();
 			Pixbuf icon = loader.get_pixbuf ();				
  			string path = this.album_art_cache_dir.concat("/downloaded-coverart-XXXXXX");

