=== modified file 'src/pidgin-libnotify.c'
Index: pidgin-libnotify-0.14/src/pidgin-libnotify.c
===================================================================
--- pidgin-libnotify-0.14.orig/src/pidgin-libnotify.c	2010-06-23 11:56:46.000000000 +0200
+++ pidgin-libnotify-0.14/src/pidgin-libnotify.c	2010-06-23 11:56:50.000000000 +0200
@@ -1151,6 +1151,20 @@
 	return FALSE;
 }
 
+static gboolean
+force_load_once (gpointer data)
+{
+	PurplePlugin * plugin = (PurplePlugin *)data;
+
+	if (!purple_prefs_get_bool("/plugins/gtk/libnotify/auto_loaded")) {
+		purple_plugin_load(plugin);
+		purple_plugins_save_loaded(PIDGIN_PREFS_ROOT "/plugins/loaded");
+		purple_prefs_set_bool("/plugins/gtk/libnotify/auto_loaded", TRUE);
+	}
+
+	return FALSE;
+}
+
 static void
 notify_check_caps_helper (gpointer data, gpointer user_data)
 {
@@ -1371,6 +1385,9 @@
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/signon", TRUE);
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/signoff", FALSE);
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/only_available", FALSE);
+	purple_prefs_add_bool ("/plugins/gtk/libnotify/auto_loaded", FALSE);
+
+	g_idle_add(force_load_once, plugin);
 }
 
 PURPLE_INIT_PLUGIN(notify, init_plugin, info)
