diff -Naur pidgin-libnotify-0.14.orig/configure.ac pidgin-libnotify-0.14/configure.ac
--- pidgin-libnotify-0.14.orig/configure.ac	2008-12-14 18:25:03.000000000 +0100
+++ pidgin-libnotify-0.14/configure.ac	2011-03-15 05:35:54.720232407 +0100
@@ -68,7 +68,7 @@
 # Check for libnotify
 #
 
-PKG_CHECK_MODULES(LIBNOTIFY, libnotify >= 0.3.2)
+PKG_CHECK_MODULES(LIBNOTIFY, libnotify >= 0.7.0)
 
 AC_SUBST(LIBNOTIFY_CFLAGS)
 AC_SUBST(LIBNOTIFY_LIBS)
diff -Naur pidgin-libnotify-0.14.orig/src/pidgin-libnotify.c pidgin-libnotify-0.14/src/pidgin-libnotify.c
--- pidgin-libnotify-0.14.orig/src/pidgin-libnotify.c	2008-12-14 18:45:51.000000000 +0100
+++ pidgin-libnotify-0.14/src/pidgin-libnotify.c	2011-03-15 05:34:37.299346231 +0100
@@ -286,7 +286,7 @@
 		g_free (tr_body);
 		return;
 	}
-	notification = notify_notification_new (title, tr_body, NULL, NULL);
+	notification = notify_notification_new (title, tr_body, NULL);
 	purple_debug_info (PLUGIN_ID, "notify(), new: "
 					 "title: '%s', body: '%s', buddy: '%s'\n",
 					 title, tr_body, best_name (buddy));
