--- xnoise-0.1.29/libxnoise/Makefile.in.g_const_return	2011-10-03 00:54:21.000000000 +0200
+++ xnoise-0.1.29/libxnoise/Makefile.in	2011-10-05 14:13:06.801576638 +0200
@@ -461,7 +461,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/src/Makefile.in.g_const_return	2011-10-03 00:54:26.000000000 +0200
+++ xnoise-0.1.29/src/Makefile.in	2011-10-05 14:13:06.799576628 +0200
@@ -313,7 +313,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/soundmenu2/Makefile.in.g_const_return	2011-10-03 00:54:25.000000000 +0200
+++ xnoise-0.1.29/plugins/soundmenu2/Makefile.in	2011-10-05 14:13:06.804576653 +0200
@@ -342,7 +342,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/soundmenu/Makefile.in.g_const_return	2011-10-03 00:54:25.000000000 +0200
+++ xnoise-0.1.29/plugins/soundmenu/Makefile.in	2011-10-05 14:13:06.810576684 +0200
@@ -343,7 +343,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/lastfmCovers/Makefile.in.g_const_return	2011-10-03 00:54:23.000000000 +0200
+++ xnoise-0.1.29/plugins/lastfmCovers/Makefile.in	2011-10-05 14:13:06.802576643 +0200
@@ -344,7 +344,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/chartlyrics/Makefile.in.g_const_return	2011-10-03 00:54:22.000000000 +0200
+++ xnoise-0.1.29/plugins/chartlyrics/Makefile.in	2011-10-05 14:13:06.802576643 +0200
@@ -343,7 +343,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     -I$(top_srcdir)/libxnoise \
--- xnoise-0.1.29/plugins/testplugin/Makefile.in.g_const_return	2011-10-03 00:54:26.000000000 +0200
+++ xnoise-0.1.29/plugins/testplugin/Makefile.in	2011-10-05 14:13:06.808576674 +0200
@@ -340,7 +340,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     -I$(top_srcdir)/libxnoise \
--- xnoise-0.1.29/plugins/titleToDecoration/Makefile.in.g_const_return	2011-10-03 00:54:26.000000000 +0200
+++ xnoise-0.1.29/plugins/titleToDecoration/Makefile.in	2011-10-05 14:13:06.805576659 +0200
@@ -343,7 +343,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/databaseLyrics/Makefile.in.g_const_return	2011-10-03 00:54:22.000000000 +0200
+++ xnoise-0.1.29/plugins/databaseLyrics/Makefile.in	2011-10-05 14:13:06.808576674 +0200
@@ -341,7 +341,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/mediakeys/Makefile.in.g_const_return	2011-10-03 00:54:24.000000000 +0200
+++ xnoise-0.1.29/plugins/mediakeys/Makefile.in	2011-10-05 14:13:06.803576648 +0200
@@ -345,7 +345,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/mpris/Makefile.in.g_const_return	2011-10-03 00:54:24.000000000 +0200
+++ xnoise-0.1.29/plugins/mpris/Makefile.in	2011-10-05 14:13:06.806576664 +0200
@@ -339,7 +339,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/cyclic-save-state/Makefile.in.g_const_return	2011-10-03 00:54:22.000000000 +0200
+++ xnoise-0.1.29/plugins/cyclic-save-state/Makefile.in	2011-10-05 14:13:06.810576684 +0200
@@ -342,7 +342,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/lyricwiki/Makefile.in.g_const_return	2011-10-03 00:54:23.000000000 +0200
+++ xnoise-0.1.29/plugins/lyricwiki/Makefile.in	2011-10-05 14:13:06.806576664 +0200
@@ -344,7 +344,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/mpris_one/Makefile.in.g_const_return	2011-10-03 00:54:24.000000000 +0200
+++ xnoise-0.1.29/plugins/mpris_one/Makefile.in	2011-10-05 14:16:08.544531576 +0200
@@ -342,7 +342,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/notifications/Makefile.in.g_const_return	2011-10-03 00:54:24.000000000 +0200
+++ xnoise-0.1.29/plugins/notifications/Makefile.in	2011-10-05 14:13:06.807576669 +0200
@@ -343,7 +343,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
--- xnoise-0.1.29/plugins/notifications2/Makefile.in.g_const_return	2011-10-03 00:54:25.000000000 +0200
+++ xnoise-0.1.29/plugins/notifications2/Makefile.in	2011-10-05 14:13:06.809576679 +0200
@@ -343,7 +343,6 @@
     -DGIO_DISABLE_DEPRECATED=1 \
     -DGDK_DISABLE_DEPRECATED=1 \
     -DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-    -DG_DISABLE_DEPRECATED=1 \
     -DPANGO_DISABLE_DEPRECATED=1 \
     -DGSEAL_ENABLE=1 \
     $(NULL)
