diff --git a/src/PdfMod/Gui/Client.cs b/src/PdfMod/Gui/Client.cs
index 19e0af6..d7a408b 100644
--- a/src/PdfMod/Gui/Client.cs
+++ b/src/PdfMod/Gui/Client.cs
@@ -415,7 +415,7 @@ namespace PdfMod.Gui
         static void OnLogNotify (LogNotifyArgs args)
         {
             ThreadAssist.ProxyToMain (delegate {
-                Gtk.MessageType mtype;
+                Gtk.MessageType mtype = Gtk.MessageType.Error;
                 var entry = args.Entry;
 
                 switch (entry.Type) {
