Index: poppler-sharp/sources/poppler-sharp.pc.in
===================================================================
--- poppler-sharp.orig/sources/poppler-sharp.pc.in	2010-06-16 18:54:25.898048663 +0800
+++ poppler-sharp/sources/poppler-sharp.pc.in	2010-06-16 18:54:42.698062124 +0800
@@ -2,7 +2,7 @@
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
-Libraries=${libdir}/@PACKAGE@/poppler-sharp.dll
+Libraries=${libdir}/@PACKAGE@/poppler-sharp.dll ${libdir}/@PACKAGE@/poppler-sharp.dll.config
 
 Name: poppler-sharp
 Description: poppler-sharp - managed CLI bindings for libpoppler
