Index: poppler-sharp/sources/poppler-sharp.pc.in
===================================================================
--- poppler-sharp.orig/sources/poppler-sharp.pc.in	2010-06-18 02:58:23.143011101 +0800
+++ poppler-sharp/sources/poppler-sharp.pc.in	2010-06-18 02:58:46.073034223 +0800
@@ -6,4 +6,5 @@
 
 Name: poppler-sharp
 Description: poppler-sharp - managed CLI bindings for libpoppler
+Libs: -r:${libdir}/@PACKAGE@/poppler-sharp.dll
 Version: @VERSION@
