diff -Naur libdbusmenu-0.3.16.orig/libdbusmenu-gtk/Makefile.am libdbusmenu-0.3.16/libdbusmenu-gtk/Makefile.am
--- libdbusmenu-0.3.16.orig/libdbusmenu-gtk/Makefile.am	2010-09-22 16:10:41.000000000 +0200
+++ libdbusmenu-0.3.16/libdbusmenu-gtk/Makefile.am	2011-03-17 05:27:33.815447701 +0100
@@ -48,7 +48,9 @@
 INTROSPECTION_SCANNER_ARGS = \
 	--add-include-path=$(srcdir) \
 	--add-include-path=$(top_builddir)/libdbusmenu-glib \
-	$(addprefix --c-include=libdbusmenu-gtk/, $(introspection_sources))
+	$(addprefix --c-include=libdbusmenu-gtk/, $(introspection_sources)) \
+	--symbol-prefix=dbusmenu \
+	--identifier-prefix=DbusmenuGtk
 INTROSPECTION_COMPILER_ARGS = --includedir=$(builddir) --includedir=$(top_builddir)/libdbusmenu-glib
 
 if HAVE_INTROSPECTION
diff -Naur libdbusmenu-0.3.16.orig/libdbusmenu-gtk/Makefile.in libdbusmenu-0.3.16/libdbusmenu-gtk/Makefile.in
--- libdbusmenu-0.3.16.orig/libdbusmenu-gtk/Makefile.in	2010-09-22 17:13:17.000000000 +0200
+++ libdbusmenu-0.3.16/libdbusmenu-gtk/Makefile.in	2011-03-17 05:27:51.093287342 +0100
@@ -331,7 +331,9 @@
 INTROSPECTION_SCANNER_ARGS = \
 	--add-include-path=$(srcdir) \
 	--add-include-path=$(top_builddir)/libdbusmenu-glib \
-	$(addprefix --c-include=libdbusmenu-gtk/, $(introspection_sources))
+	$(addprefix --c-include=libdbusmenu-gtk/, $(introspection_sources)) \
+	--symbol-prefix=dbusmenu \
+	--identifier-prefix=DbusmenuGtk
 
 INTROSPECTION_COMPILER_ARGS = --includedir=$(builddir) --includedir=$(top_builddir)/libdbusmenu-glib
 @HAVE_INTROSPECTION_TRUE@introspection_sources = $(libdbusmenu_gtkinclude_HEADERS)
