Index: pino-0.2.10/src/account_action.vala
===================================================================
--- pino-0.2.10.orig/src/account_action.vala
+++ pino-0.2.10/src/account_action.vala
@@ -38,7 +38,8 @@ public class AccountAction : Action {
 	weak Thread thread_1;
 	
 	public AccountAction() {
-		name = "AccountAct";
+		GLib.Object (name: "AccountAct");
+
 		label = _("Accounts");
 		tooltip = null;
 		stock_id = null;
Index: pino-0.2.10/src/more_window.vala
===================================================================
--- pino-0.2.10.orig/src/more_window.vala
+++ pino-0.2.10/src/more_window.vala
@@ -28,7 +28,7 @@ public class MoreWindow : Window {
 	public signal void moar_event();
 	
 	public MoreWindow() {
-		type = WindowType.POPUP;
+		GLib.Object (type: WindowType.POPUP);
 		
 		//set_position(WindowPosition.MOUSE);
 		
--- pino-0.2.10/src/gtk_style.vala	2010-05-14 12:38:02.000000000 +0200
+++ pino-0.2.10/src/gtk_style.vala.new	2011-02-28 12:03:49.951165086 +0100
@@ -42,7 +42,7 @@
 		sl_color = rgb_to_hex(style.bg[Gtk.StateType.SELECTED]);
 		
 		Value v = new Value(typeof(Gdk.Color));// = null;
-		style.get_style_property(typeof(Gtk.Widget), "link-color", v);
+		style.get_style_property(typeof(Gtk.Widget), "link-color", out v);
 		lk_color = rgb_to_hex((Gdk.Color) v);
 		
 		//working on light color (lt_color)
