diff -up glew-1.5.5/config/Makefile.linux.mx glew-1.5.5/config/Makefile.linux
--- glew-1.5.5/config/Makefile.linux.mx	2010-12-16 23:06:16.169124367 +0000
+++ glew-1.5.5/config/Makefile.linux	2010-12-16 23:07:07.012627069 +0000
@@ -3,6 +3,7 @@ CC = cc
 LD = cc
 ifneq (undefined, $(origin GLEW_MX))
 CFLAGS.EXTRA = -DGLEW_MX
+NAME_SUF=mx
 endif
 PICFLAG = -fPIC
 LDFLAGS.SO = -shared -Wl,-soname=$(LIB.SONAME)
@@ -17,7 +18,7 @@ endif
 LDFLAGS.GL = -lGL
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
-NAME = GLEW
+NAME = GLEW$(NAME_SUF)
 WARN = -Wall -W
 POPT = -O2
 BIN.SUFFIX =
diff -up glew-1.5.5/Makefile.mx glew-1.5.5/Makefile
--- glew-1.5.5/Makefile.mx	2010-12-16 22:55:34.279026158 +0000
+++ glew-1.5.5/Makefile	2010-12-16 23:16:01.550839987 +0000
@@ -80,7 +80,8 @@ VISUALINFO.BIN.SRCS = src/visualinfo.c
 VISUALINFO.BIN.OBJS = $(VISUALINFO.BIN.SRCS:.c=.o)
 BIN.LIBS = -Llib $(LDFLAGS.DYNAMIC) -l$(NAME) $(LDFLAGS.EXTRA) $(LDFLAGS.GL) -lX11
 
-all debug: lib/$(LIB.SHARED) lib/$(LIB.STATIC) bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) glew.pc
+all debug: lib/$(LIB.SHARED) lib/$(LIB.STATIC) bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) glew.pc glewmx.pc
+libsonly: lib/$(LIB.SHARED) lib/$(LIB.STATIC) glew.pc glewmx.pc
 
 lib:
 	mkdir lib
@@ -120,9 +121,17 @@
 		-e "s|@version@|$(GLEW_VERSION)|g" \
 		< $< > $@
 
-install: all
+glewmx.pc: glewmx.pc.in
+	sed \
+		-e "s|@prefix@|$(GLEW_DEST)|g" \
+		-e "s|@libdir@|$(LIBDIR)|g" \
+		-e "s|@exec_prefix@|$(BINDIR)|g" \
+		-e "s|@includedir@|$(INCDIR)|g" \
+		-e "s|@version@|$(GLEW_VERSION)|g" \
+		< $< > $@
+
+install-libs: libsonly
 # directories
-	$(INSTALL) -d -m 0755 $(BINDIR)
 	$(INSTALL) -d -m 0755 $(INCDIR)
 	$(INSTALL) -d -m 0755 $(LIBDIR)
 	$(INSTALL) -d -m 0755 $(LIBDIR)/pkgconfig
@@ -139,7 +148,11 @@
 	$(INSTALL) -m 0644 include/GL/wglew.h $(INCDIR)/
 	$(INSTALL) -m 0644 include/GL/glew.h $(INCDIR)/
 	$(INSTALL) -m 0644 include/GL/glxew.h $(INCDIR)/
+ifneq (undefined, $(origin GLEW_MX))
+	$(INSTALL) -m 0644 glewmx.pc $(LIBDIR)/pkgconfig/
+else
 	$(INSTALL) -m 0644 glew.pc $(LIBDIR)/pkgconfig/
+endif
 ifeq ($(filter-out mingw% cygwin,$(SYSTEM)),)
 	$(INSTALL) -m 0644 lib/$(LIB.DEVLNK) $(LIBDIR)/
 else
@@ -147,9 +159,15 @@
 	$(INSTALL) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.DEVLNK)
 endif
+
+install-bin:
+# directories
+	$(INSTALL) -d -m 0755 $(BINDIR)
 # utilities
 	$(INSTALL) -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
 
+install: all install-libs install-bin
+
 uninstall:
 	$(RM) $(INCDIR)/wglew.h
 	$(RM) $(INCDIR)/glew.h
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ glew-1.5.5/glewmx.pc.in	2010-05-21 20:25:40.000000000 +0200
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: glewmx
+Description: The OpenGL Extension Wrangler library (mx)
+Version: @version@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lGLEWmx
