=== modified file 'libindicate/server.c'
--- old/libindicate/server.c	2010-09-03 17:48:41 +0000
+++ new/libindicate/server.c	2010-12-13 20:35:08 +0000
@@ -1156,7 +1156,7 @@
 	use this function to create a #IndicateServer as it ensure that there
 	is only one per application.
 
-	Return value: A reference to the default #IndicateServer instance.
+	Return value: (transfer none): A reference to the default #IndicateServer instance.
 */
 IndicateServer *
 indicate_server_ref_default (void)

=== modified file 'libindicate/server.h'
--- old/libindicate/server.h	2010-06-04 19:28:17 +0000
+++ new/libindicate/server.h	2010-12-13 20:40:32 +0000
@@ -233,6 +233,17 @@
 void indicate_server_add_indicator (IndicateServer * server, IndicateIndicator * indicator);
 void indicate_server_remove_indicator (IndicateServer * server, IndicateIndicator * indicator);
 
+/**
+ * indicate_server_ref_default:
+
+ * This function will return a reference to the default #IndicateServer
+ * reference if there is one, or it will create one if one had not
+ * previously been created.  It is recommended that all applications
+ * use this function to create a #IndicateServer as it ensure that there
+ * is only one per application.
+ * 
+ * Return value: (transfer none): A reference to the default #IndicateServer instance.
+*/
 IndicateServer * indicate_server_ref_default (void);
 void indicate_server_set_default (IndicateServer * server);
 

