diff -Naur goobox-2.2.0.orig/configure.ac goobox-2.2.0/configure.ac
--- goobox-2.2.0.orig/configure.ac	2010-11-15 21:23:00.000000000 +0100
+++ goobox-2.2.0/configure.ac	2011-03-15 01:42:03.023137528 +0100
@@ -18,7 +18,7 @@
 GLIB_REQUIRED=2.22
 GTK_REQUIRED=2.18
 GSTREAMER_REQUIRED=0.10.12
-LIBNOTIFY_REQUIRED=0.4.3
+LIBNOTIFY_REQUIRED=0.7.0
 LIBMUSICBRAINZ_REQUIRED=2.1.0
 
 dnl ===========================================================================
diff -Naur goobox-2.2.0.orig/src/main.c goobox-2.2.0/src/main.c
--- goobox-2.2.0.orig/src/main.c	2010-07-30 18:53:48.000000000 +0200
+++ goobox-2.2.0/src/main.c	2011-03-15 01:41:28.142103266 +0100
@@ -591,7 +591,7 @@
 			g_list_free (caps);
 		}
 
-		notification = notify_notification_new_with_status_icon (title, msg, "goobox", status_icon);
+		notification = notify_notification_new (title, msg, NULL);
 		notify_notification_set_urgency (notification, NOTIFY_URGENCY_LOW);
 
 		if (supports_actions) {
@@ -612,11 +612,6 @@
 	else
 		notify_notification_update (notification, title, msg, "goobox");
 
-	/*if ((x >= 0) && (y >= 0))
-		notify_notification_set_geometry_hints (notification,
-							screen,
-							x, y);*/
-
 	notify_notification_show (notification, NULL);
 	
 #endif /* ENABLE_NOTIFICATION */
